addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1079 +1,1049 @@
+                elements: this._getElementOptions(),
+            };
+            if (this._isRedirectionEnabled()) {
+                options.onClick = this._onGraphClicked.bind(this);
             }
-            return labelCopy.join('/');
+            return options;
         }
-        // here label is an array containing strings or numbers.
-        return label.join('/') || _t('Total');
-    },
-    /**
-     * Render the chart or display a message error in case data is not good enough.
-     *
-     * Note that This method is synchronous, but the actual rendering is done
-     * asynchronously.  The reason for that is that Chart.js needs to be in the
-     * DOM to correctly render itself.  So, we trick Odoo by returning
-     * immediately, then we render the chart when the widget is in the DOM.
-     *
-     * @override
-     * @private
-     * @returns {Promise} The _super promise is actually resolved immediately
-     */
-    _render: function () {
-        if (this.chart) {
-            this.chart.destroy();
+
+        /**
+         * Determine how to relabel a label according to a given origin.
+         * The idea is that the getLabel function is in general not invertible but
+         * it is when restricted to the set of dataPoints coming from a same origin.
+
+         * @private
+         * @param {Array} label
+         * @param {Array} originIndex
+         * @returns {string}
+         */
+        _relabelling(label, originIndex) {
+            if (label.isNoData) {
+                return label[0];
+            }
+            const i = this.props.comparisonFieldIndex;
+            if (['bar', 'line'].includes(this.props.mode) && i === 0) {
+                // here label is an array of length 1 and contains a number
+                return this.dateClasses.representative(label, originIndex) || '';
+            } else if (this.props.mode === 'pie' && i === 0) {
+                // here label is an array of length at least one containing string or numbers
+                const labelCopy = label.slice(0);
+                if (originIndex !== undefined) {
+                    labelCopy.splice(i, 1, this.dateClasses.representative(label[i], originIndex));
+                } else {
+                    labelCopy.splice(i, 1, this.dateClasses.dateClassMembers(label[i]));
+                }
+                return labelCopy.join('/');
+            }
+            // here label is an array containing strings or numbers.
+            return label.join('/') || _t('Total');
         }
-        this.$el.empty();
-        if (!_.contains(CHART_TYPES, this.state.mode)) {
-            this.trigger_up('warning', {
-                title: _t('Invalid mode for chart'),
-                message: _t('Cannot render chart with mode : ') + this.state.mode
+
+        /**
+         * create bar chart.
+         *
+         * @private
+         * @param {Object[]} dataPoints
+         */
+        _renderBarChart(dataPoints) {
```

Comment:
dataPoints is this.filteredDataPoints, right? We could change signature and use that

Reply:
same for other _render... methods

Reply:
Would you do a `const dataPoints = this.filteredDataPoints` in the beginning of the methods, or replace all `dataPoints` with `this.filteredDataPoints`?